packages/text-field/src/TextField.test.js

```diff
@@ -4,6 +4,18 @@ import TextField from "./index";
 import Input from "./presenters/Input";
 
 describe("TextField", () => {
+  it("allows value to be changed only by onChange", () => {
+    const wrapper = mount(<TextField value="foo" />);
+    const input = wrapper.find("input");
+
+    expect(input.prop("value")).toEqual("foo");
+    input.simulate("change", { target: { value: "bar" } });
+    expect(input.prop("value")).toEqual("bar");
+
+    wrapper.setProps({ value: "baz" });
+    expect(input.prop("value")).toEqual("bar"); // Incoming props are ignored
```

Comment:
Lets move this into a separate test `it("ignores incoming value prop changes", ...)`